### PR TITLE
[enhancement](memory) Add tablet schema cache metrics

### PR DESCRIPTION
**Pull request by xinyiZzz**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.16 seconds
 load time: 438 seconds
 storage size: 17123343043 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221201164330_clickbench_pr_56343.html

File: be/src/olap/tablet_schema_cache.h [hunk 1]
```diff
@@ -49,6 +50,9 @@ class TabletSchemaCache {
             pb.ParseFromString(key);
             tablet_schema_ptr->init_from_pb(pb);
             _cache[key] = tablet_schema_ptr;
+            DorisMetrics::instance()->tablet_schema_cache_count->increment(1);
+            DorisMetrics::instance()->tablet_schema_cache_memory_bytes->increment(
+                    tablet_schema_ptr->mem_size());
```
Comment (yiguolei) [inline be/src/olap/tablet_schema_cache.h:55]: mem_size method is not accurate. And I find it is very tick.
   std::vector<TabletColumn> _cols;
    std::vector<TabletIndex> _indexes;
    std::unordered_map<std::string, int32_t> _field_name_to_index;
    std::unordered_map<int32_t, int32_t> _field_id_to_index;

TableSchema has a lot of properties. 

I think you could add a property in TabletSchema and using scoped_mem_tracker to set its size during init_from_pb.

And you could remove all mem_size method. I find it is useless before.

Comment (xinyiZzz) [reply be/src/olap/tablet_schema_cache.h:55]: good suggestion, fixed

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**